src/main/kotlin/com/networkedassets/gherkin/runner/util/GherkinConverter.kt

```diff
@@ -64,6 +63,19 @@ object GherkinConverter {
         return gherkinScenario
     }
 
+    private fun convert2DArrToEnvBindings(dataTableForStep: Array<Array<String>>?): Map<String, String> {
+        val converted = mutableMapOf<String, String>()
+        dataTableForStep
+                ?.dropWhile { row -> row[0].contains("#") }
+                ?.map { row -> converted.put(row[0], row[1]) }
```

Comment:
Using map for mutating object, which is out of scope of mapping function is bad idea, which is misleading. You should use foreach in this case. There is also other approach. You can use associate() method to map list to Map and then assign it to "converted" variable:
.associate{ row  -> row[0] to row[1] }